### PR TITLE
PATCH api/product/:productId/requesters/:requesterId/approve

### LINKED ISSUE #45: PATCH /products/:productId/requesters/:requesterId/approve
First of all, check if the requesterId is in the `orderRequests`. 
Then you are going to assign the user as the `beneficiary`.

Private endpoint.
Authorization needed.

Create swagger documentation.

### PR DESCRIPTION
**Pull request by awiednoor**
- Approve request of the product
- checks if the requesterID is in the orderRequests
- Assigns requesterID to the beneficiary
- Updates the user model of the requesterID to have the productID in the received array

### CODE REVIEW

File: src/controllers/product.js
Comment (okurezgi) [inline src/controllers/product.js]: maybe we can change naming as approveRequest or approveRequester, it is giving the function main focus with that kind of naming i guess

File: src/controllers/product.js
Comment (okurezgi) [inline src/controllers/product.js:330]: i think we also need to delete the product id from other requesters' requested array in user model with a loop, but if it is too much work or too complicated for now, we can skip that it is up to you

File: src/docs/Product/approveProductRequest.yaml
Comment (okurezgi) [inline src/docs/Product/approveProductRequest.yaml]: there is a typo in updates

File: src/docs/Product/approveProductRequest.yaml
Comment (okurezgi) [inline src/docs/Product/approveProductRequest.yaml]: shouldn't this be same with the function name?

File: src/docs/Product/approveProductRequest.yaml
Comment (okurezgi) [inline src/docs/Product/approveProductRequest.yaml]: this example is a product id, you can change it with this one, (owner of the product id above): 62e2f022552347123617a73f

**Review by okurezgi — CHANGES_REQUESTED** [review]
Noor great code, i just added some comments, let me know when you need another review

File: src/controllers/product.js
Comment (awiednoor) [reply src/controllers/product.js]: Ok changed it

File: src/controllers/product.js
Comment (awiednoor) [reply src/controllers/product.js:330]: Yes it can be done , using mongoose methods without a loop

File: src/docs/Product/approveProductRequest.yaml
Comment (awiednoor) [reply src/docs/Product/approveProductRequest.yaml]: Done

File: src/controllers/product.js
Comment (awiednoor) [reply src/controllers/product.js:330]: Done

Comment (okurezgi) [reply src/controllers/product.js:330]: no no i meant also from the other requesters' array the ones that are not approved, but as i said before, not that important rn so we can move on

File: src/docs/Product/approveProductRequest.yaml
Comment (okurezgi) [reply src/docs/Product/approveProductRequest.yaml]: examples should be visa versa 62e2f022552347123617a73f is a user id, 62f661cf728303c5bead3b6d is a product id

File: src/docs/Product/approveProductRequest.yaml
Comment (okurezgi) [inline src/docs/Product/approveProductRequest.yaml]: i think it is intended to write product of requester, there is a typo here

**Review by okurezgi — CHANGES_REQUESTED** [review]

File: src/controllers/product.js
Comment (awiednoor) [reply src/controllers/product.js:330]: I wrote it like that the first time around, but then I figured it's not important either way if the transaction is completed you can't edit the requesters .
I thought i would leave it to know that this requester was in the list.

**Review by okurezgi — APPROVED** [review]

**Review by halitbatur — APPROVED** [review]

**Pull request merged**
